cmd/snap/cmd_model.go

```diff
@@ -304,6 +310,75 @@ func (x *cmdModel) Execute(args []string) error {
 					fmt.Fprintln(w, "device-key-sha3-384: |")
 					wrapLine(w, []rune(headerString), "  ", termWidth)
 				}
+			case "snaps":
+				// also flush the writer before continuing so the previous keys
+				// don't try to align with this key
+				w.Flush()
+				snapsHeader, ok := headerValue.([]interface{})
+				if !ok {
+					return invalidTypeErr
+				}
+				if len(snapsHeader) == 0 {
+					// unexpected why this is an empty list, but just ignore for
+					// now
+					continue
+				}
+				fmt.Fprintf(w, "snaps:\n")
+				for _, sn := range snapsHeader {
+					snMap, ok := sn.(map[string]interface{})
+					if !ok {
+						return invalidTypeErr
+					}
+					// iterate over all keys in the map in a stable, visually
+					// appealing ordering
+					// first do snap name, which will always be present
+					name, ok := snMap["name"]
+					if !ok {
+						// TODO: more specific error ?
+						return invalidTypeErr
+					}
+					nameStr, ok := name.(string)
+					if !ok {
+						return invalidTypeErr
+					}
+					fmt.Fprintf(w, "  - name:\t%s\n", nameStr)
+
+					// the rest of these may be absent,
+					for _, snKey := range []string{"type", "default-channel", "id"} {
```

Comment:
there's also "presence", I would go with:

{"id", "type", "default-channel", "presence"}

Reply:
A great candidate for bikeshedding. How about type, id, channel, presence?
```
- name: snapd
  type: snapd
  id: 12341234
- name: pc
  type: gadget
  id: 23452345abc
  default-channel: 20/stable
- name: pc-kernel
  type: kernel
  id: abcdef1234
  default-channel: 20/stable
```

vs 

```
- name: snapd
  id: 12341234
  type: snapd
- name: pc
  id: 23452345abc
  type: gadget
  default-channel: 20/stable
- name: pc-kernel
  id: abcdef1234
  type: kernel
  default-channel: 20/stable
```

Reply:
I still prefer the latter

Reply:
changed